domain/application/service/application.go

```diff
@@ -1570,17 +1602,11 @@ func (s *Service) GetApplicationStatus(ctx context.Context, appID coreapplicatio
 		return nil, internalerrors.Capture(err)
 	}
 
-	deocodedStatuses := []corestatus.StatusInfo{}
-	for _, unitStatus := range unitStatuses {
-		status, err := decodeWorkloadStatus(&unitStatus)
-		if err != nil {
-			return nil, internalerrors.Capture(err)
-		}
-		deocodedStatuses = append(deocodedStatuses, *status)
+	derivedApplicationStatus, err := reduceWorkloadStatuses(slices.Collect(maps.Values(unitStatuses)))
```

Comment:
instead of having `reduceWorkloadStatuses()` you could have only a compare function and then this becomes:
```suggestion
	derivedApplicationStatus := slices.SortFunc(maps.Values(unitStatuses), compareFunc)[0]
```

Reply:
Honestly I do not mind either way. I don't see any clear advantage to either way